src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,106 @@
+package duke;
+
+import java.util.ArrayList;
+
+import task.Task;
+
+public class TaskList {
+
+    private ArrayList<Task> tasks;
+
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Adds Task to task list.
+     *
+     * @param t Task to add to task list.
+     */
+    public String add(Task t) {
+        this.tasks.add(t);
+        return "Added task: " + t.toString() + "\nYou have " + tasks.size() + " tasks in the list";
+    }
+
+    /**
+     * Adds Task to task list.
+     *
+     * @param s String stored in storage.
+     */
+    public void add(String s) {
+        this.tasks.add(Task.getTask(s));
+    }
+
+    /**
+     * Prints all the Tasks in task list.
+     */
+    public String printTasks() {
+        return this.toString();
+    }
+
+    /**
+     * Marks Task as done.
+     *
+     * @param id Id of tasks done.
+     * @throws DukeException if Id is invalid.
+     */
+    public void doneTask(int id) throws DukeException {
+        if (id >= this.tasks.size()) {
+            throw new DukeException("no such task");
+        }
+        Task t = this.tasks.get(id);
+        t.markDone();
+    }
+
+    /**
+     * Deletes Task from task list.
+     *
+     * @param id Id of task to delete.
+     * @throws DukeException if Id is invalid.
+     */
+    public void deleteTask(int id) throws DukeException {
+        if (id >= this.tasks.size()) {
+            throw new DukeException("no such task");
+        }
+        this.tasks.remove(id);
+    }
+
+    /**
+     * Finds Task that contains keyword.
+     *
+     * @param keyword Keyword used to find Task.
+     */
+    public String findTasks(String keyword) {
+        StringBuilder response = new StringBuilder("Matching task(s) are:");
+        for (int i = 0; i < tasks.size(); i++) {
+            Task t = tasks.get(i);
+            String name = t.getName();
+            if (name.contains(keyword)) {
+                response.append(i).append(": ").append(t.toString());
+            }
+        }
+        return response.toString();
+    }
+
+    /**
+     * Returns string representing task list to be saved to text file.
+     *
+     * @return String representing task list
+     */
+    public String saveTasklist() {
+        String txt = "";
+        for (int i = 0; i < tasks.size(); i++) {
+            txt = txt + tasks.get(i).saveTask() + "\n";
+        }
+        return txt;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder out = new StringBuilder();
+        for (int i = 0; i < this.tasks.size(); i++) {
+            out.append(i).append(": ").append(this.tasks.get(i).toString()).append("\n");
```

Comment:
This line is a bit too long and not easy to understand. Maybe we can consider separate it.